bundles/org.openhab.binding.entsoe/src/main/java/org/openhab/binding/entsoe/internal/EntsoeHandler.java

```diff
@@ -0,0 +1,265 @@
+
+        config = getConfigAs(EntsoeConfiguration.class);
+
+        ZonedDateTime startUtc = currentUtcTimeWholeHours()
+                .minusDays(needToFetchHistoricDays() ? config.historicDays : 1).withHour(22);
+        ZonedDateTime endUtc = currentUtcTimeWholeHours().plusDays(1).withHour(22);
+
+        boolean needsUpdate = lastDayAheadReceived.equals(ZonedDateTime.of(LocalDateTime.MIN, ZoneId.of("UTC")))
+                || responseMap.isEmpty() || needToFetchHistoricDays(true);
+
+        boolean hasNextDayValue = needsUpdate ? false
+                : responseMap.entrySet().stream()
+                        .anyMatch(x -> x.getKey().getDayOfYear() == currentUtcTime().plusDays(1).getDayOfYear());
+        boolean readyForNextDayValue = needsUpdate ? true
+                : currentUtcTime().isAfter(currentUtcTime().withHour(config.spotPricesAvailableUtcHour));
+
+        // Update whole time series
+        if (needsUpdate || (!hasNextDayValue && readyForNextDayValue)) {
+            logger.debug("Updating timeseries");
+            Request request = new Request(config.securityToken, config.area, startUtc, endUtc);
+            Client client = new Client();
+            boolean success = false;
+
+            try {
+                responseMap = client.doGetRequest(request, 30000);
+                TimeSeries baseTimeSeries = new TimeSeries(EntsoeBindingConstants.TIMESERIES_POLICY);
+                for (Map.Entry<ZonedDateTime, Double> entry : responseMap.entrySet()) {
+                    BigDecimal kwhPrice = BigDecimal.valueOf(entry.getValue()).divide(new BigDecimal(1000), 10,
+                            RoundingMode.HALF_UP);
+                    baseTimeSeries.add(entry.getKey().toInstant(), getPriceState(kwhPrice));
+                }
+                lastDayAheadReceived = currentUtcTime();
+                sendTimeSeries(EntsoeBindingConstants.CHANNEL_SPOT_PRICES, baseTimeSeries);
+                updateCurrentHourState(EntsoeBindingConstants.CHANNEL_SPOT_PRICES);
+                scheduler.schedule(this::triggerChannelSpotPricesReceived, 0, TimeUnit.SECONDS);
+                success = true;
+            } catch (EntsoeResponseException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+            } catch (EntsoeConfigurationException e) {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            } finally {
+                schedule(success);
+            }
+        } else {
+            logger.debug("Updating current hour");
+            updateCurrentHourState(EntsoeBindingConstants.CHANNEL_SPOT_PRICES);
+            schedule(true);
+        }
+    }
+
+    private State getPriceState(BigDecimal kwhPrice) {
+        if (baseUnit == null || fromUnit == null) {
+            return new DecimalType(kwhPrice);
+        }
+        return new QuantityType<>(kwhPrice + " " + fromUnit.getName() + "/kWh");
```

Comment:
This can throw `IllegalArgumentException` if EUR is not known because of unit configuration.

See for example:
https://github.com/openhab/openhab-addons/blob/b77458dca7b092ed0b7740240d4fe72aebb3415f/bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/handler/EnergiDataServiceHandler.java#L502-L512

Reply:
Confirmed by test when having `LocalBasedCurrencyProvider` configured and country set to Denmark:
```
2024-09-22 00:19:06.739 [ERROR] [ore.common.registry.AbstractRegistry] - Cannot inform the listener "org.openhab.core.thing.internal.ChannelLinkNotifier@153482c4" about the "ADDED" event: Invalid Quantity value: 0.0882900000 EUR/kWh
java.lang.IllegalArgumentException: Invalid Quantity value: 0.0882900000 EUR/kWh
	at org.openhab.core.library.types.QuantityType.<init>(QuantityType.java:171) ~[?:?]
	at org.openhab.core.library.types.QuantityType.<init>(QuantityType.java:121) ~[?:?]
	at org.openhab.binding.entsoe.internal.EntsoeHandler.getPriceState(EntsoeHandler.java:237) ~[?:?]
	at org.openhab.binding.entsoe.internal.EntsoeHandler.refreshPrices(EntsoeHandler.java:212) ~[?:?]
	at org.openhab.binding.entsoe.internal.EntsoeHandler.channelLinked(EntsoeHandler.java:91) ~[?:?]
(...)
```